raspberry/bzzz/scheduler.py

```diff
@@ -0,0 +1,95 @@
+import bzzz.thread_this
+from time import time_ns
+
+
+class Scheduler:
+    # Dict keys
+    _function_ID = "function_ID"
+    _function_obj = "function_obj"
+    _function_call_frequency = "function_call_frequency"
+    _function_call_count = "function_call_count"
+    _function_thread = "function_thread"
+    _num_times_called = "num_times_called"
+    _time_at_last_call_ns = "time_at_last_call_ns"
+    _dummy_func = "dummy_func"
+
+    def __init__(self, use_threading=False) -> None:
+        # This variable is a dict with the following structure
+        # {function_name: {function_ID: int, function_obj: obj, function_call_frequency: Hz, function_call_count: int, function_thread: threading.Timer or None, num_times_called: int, time_at_last_call_ns: int}}
+        self.__functions_to_run = {}
+        self.__last_function_ID = -1
+        self.__use_threading = use_threading
+        self.__scheduled_events = []
+
+        self.__functions_to_run[Scheduler._dummy_func] = {
+            Scheduler._function_ID: self.__last_function_ID, Scheduler._function_obj: -1,
+            Scheduler._function_call_frequency: -1, Scheduler._function_call_count: -1,
+            Scheduler._function_thread: -1, Scheduler._num_times_called: -1, Scheduler._time_at_last_call_ns: -1}
+
+    def schedule(self, function_name: str, function_obj, function_call_frequency, function_call_count=0):
```

Comment:
Regarding `function_name: str`, I see that you have documented the datatype this way in a very few cases. Since this is not something we do uniformly throughout the code, I'd suggest that we abandon it and mention the data type in the docstrings.